kmod/src/xattr.c

```diff
@@ -498,6 +518,52 @@ static int parse_totl_u64(const char *s, int len, u64 *res)
 	return kstrtoull(str, 0, res) != 0 ? -EINVAL : 0;
 }
 
+static int parse_totl_u32(const char *s, int len, unsigned long int *res)
+{
+	char str[SCOUTFS_XATTR_MAX_TOTL_U32 + 1];
+
+	if (len <= 0 || len >= ARRAY_SIZE(str) || s[0] == '+' || s[len - 1] == '\n')
+		return -EINVAL;
+
+	memcpy(str, s, len);
+	str[len] = '\0';
+
+	return kstrtoul(str, 0, res) != 0 ? -EINVAL : 0;
+}
+
+static int parse_worm_timeval(struct scoutfs_inode_info *si,
```

Comment:
This needs to be rethought.   It can try and parse a negative length u32 outside the name/name_len bounds on a final dot, you can give a whole ton of dots and you'll get whatever's last, and if there's only dots or no name (as happens with deletion) It'll put on-stack garbage into the inode, which can then leak to disk, which is a huge no-no in the kernel.   unknown stack/memory contents can be priviledged, we have no idea and must not let them leave.  I'd refactor it into:  name_len has to be >= 3, only one dot that's not first or last, pass in struct scoutfs_time spec not si, sec and nsec before and after dot must parse.